mlrun/model.py

```diff
@@ -443,15 +446,27 @@ def with_commands(
     def with_requirements(
         self,
         requirements: Union[str, List[str]],
+        requirements_file: str = "",
         overwrite: bool = False,
     ):
         """add package requirements from file or list to build spec.
 
-        :param requirements:  python requirements file path or list of packages
-        :param overwrite:     overwrite existing requirements
+        :param requirements:        python requirements file path or list of packages
+        :param requirements_file:   a list of python packages
+        :param overwrite:           overwrite existing requirements
         :return: function object
         """
-        resolved_requirements = self._resolve_requirements(requirements)
+        if isinstance(requirements, str) and os.path.isfile(requirements):
+            # TODO: remove in 1.6.0
+            warnings.warn(
+                "Passing a requirements file path as a string in the 'requirements' argument is deprecated "
```

Comment:
so if `requirements="mlrun"` and you have `mlrun` file where you run it?

Reply:
So do we only want to check if it's specifically `requirements.txt`? or a `.txt` file?
Currently if it is a file that exists it will be read as a requirements file.

Reply:
I think so, yeah, yes extension (any)